src/main/java/org/jitsi/videobridge/octo/OctoPacket.java

```diff
@@ -0,0 +1,242 @@
+        case VIDEO:
+            return OCTO_MEDIA_TYPE_VIDEO;
+        case DATA:
+            return OCTO_MEDIA_TYPE_DATA;
+        default:
+            return -1;
+        }
+    }
+
+    /**
+     * Writes an Octo header to the specified buffer at the specified offset.
+     * @param buf the buffer to write to.
+     * @param off the offset to write at.
+     * @param r the value of the {@code r} flag.
+     * @param mediaType the media type.
+     * @param s the value of the {@code s} flag.
+     * @param conferenceId the Octo ID of the conference.
+     * @param endpointId the Octo ID of the endpoint.
+     */
+    public static void writeHeaders(byte[] buf, int off,
+                                    boolean r, MediaType mediaType, int s,
+                                    String conferenceId,
+                                    String endpointId)
+    {
+        buf[off] = 0;
+        if (r)
+        {
+            buf[off] |= 0x80;
+        }
+        buf[off] |= (getMediaTypeId(mediaType) & 0x03) << 5;
+        buf[off] |= (s & 0x03) << 3;
+
+        writeConferenceId(conferenceId, buf, off, OCTO_HEADER_LENGTH);
+        writeEndpointId(endpointId, buf, off, OCTO_HEADER_LENGTH);
+    }
+
+    /**
+     * Reads the conference ID from an Octo header.
+     * @param buf the buffer which contains the Octo header.
+     * @param off the offset in {@code buf} at which the Octo header begins.
+     * @param len the length of the buffer.
+     * @return the Octo conference ID read from the buffer.
+     */
+    public static String readConferenceId(byte[] buf, int off, int len)
+    {
+        assertMinLen(buf, off, len);
+
+        int cid = RTPUtils.readUint24AsInt(buf, off + 1);
+        return Integer.toHexString(cid);
+    }
+
+    /**
+     * Reads the {@link MediaType} from an Octo header.
+     * @param buf the buffer which contains the Octo header.
+     * @param off the offset in {@code buf} at which the Octo header begins.
+     * @param len the length of the buffer.
+     * @return the {@link MediaType} from the given Octo header.
+     */
+    public static MediaType readMediaType(byte[] buf, int off, int len)
+    {
+        assertMinLen(buf, off, len);
+
+        int mediaType = (buf[off] & 0x60) >> 5;
+        switch (mediaType)
+        {
+        case OCTO_MEDIA_TYPE_AUDIO:
+            return MediaType.AUDIO;
+        case OCTO_MEDIA_TYPE_VIDEO:
+            return MediaType.VIDEO;
+        case OCTO_MEDIA_TYPE_DATA:
+            return MediaType.DATA;
+        default:
+            return null;
+        }
+    }
+
+    /**
+     * Reads the {@code r} flag from an Octo header.
+     * @param buf the buffer which contains the Octo header.
+     * @param off the offset in {@code buf} at which the Octo header begins.
+     * @param len the length of the buffer.
+     * @return the {@code r} flag from the given Octo header.
+     */
+    public static boolean readRflag(byte[] buf, int off, int len)
```

Comment:
RawPacket?

Reply:
All the Octo stuff currently happens before the packets reach libjitsi (or after they leave libjitsi), so there's no RawPacket instance.

Reply:
i see...but is there use in changing the buffers to a RawPacket earlier (at this point?)

Reply:
I don't think so. Apart from the check for RTP vs RTCP, I don't think we will need to do any RTP or RTCP specific things with the packets in the Octo layer. Note that OctoPacket only has static methods, we don't create any instances.